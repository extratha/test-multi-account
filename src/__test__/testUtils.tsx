import { RenderOptions, RenderResult, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SWRConfig } from 'swr';
import { theme } from '@/config/config-mui/theme';
import { ThemeProvider as MUIThemeProvider } from '@mui/material/styles';
import { NextIntlClientProvider } from 'next-intl';
import commonTh from '../../public/locales/th/common.json';
import commonEn from '../../public/locales/th/common.json';
type RenderWithProvider = (
  elm: React.ReactElement,
  renderOptions?: ProviderOptions,
) => RenderResult;

export type ProviderOptions = {
  formOptions?: Record<string, unknown>;
} & RenderOptions;

const messages = {
  ...commonTh,
  ...commonEn,
};

const renderWithProviders: RenderWithProvider = (
  component,
  renderOptions?,
  locale = 'th',
) => {
  const wrapper = ({ children }: { children: React.ReactNode }) => (
    <NextIntlClientProvider locale={locale} messages={messages}>
      <MUIThemeProvider theme={theme}>
        <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
      </MUIThemeProvider>
    </NextIntlClientProvider>
  );

  return render(component, {
    wrapper,
    ...renderOptions,
  });
};

export { renderWithProviders as render, userEvent };
export * from '@testing-library/react';
