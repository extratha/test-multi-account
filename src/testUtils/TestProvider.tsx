import { ThemeProvider } from "@emotion/react";

import { ReactNode } from "react";
import { SWRConfig } from "swr";

import ModalLayer from "@/components/Modal/ModalLayer";
import { theme } from "@/config/config-mui";

interface TestProviderProps {
  children?: ReactNode;
}

export const TestProvider = ({ children }: TestProviderProps) => {
  return (
    <ThemeProvider theme={theme}>
      <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
      <ModalLayer />
    </ThemeProvider>
  );
};
