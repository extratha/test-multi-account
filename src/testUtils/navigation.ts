import * as NextNavigation from "next/navigation";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
  useParams: jest.fn(),
  usePathname: jest.fn(),
  useSearchParams: jest.fn(),
  useServerInsertedHTML: jest.fn(),
}));

export interface SpyUseSearchParams {
  get: jest.Mock;
}

export interface SpyUseRouter {
  replace: jest.Mock;
  push: jest.Mock;
  back: jest.Mock;
}

export const spyUseParams = (params: any = {}) => {
  jest.spyOn(NextNavigation, "useParams").mockReturnValue(params);
  return NextNavigation.useParams as jest.Mock;
};

export const spyUseRouter = (): SpyUseRouter => {
  const replace = jest.fn();
  const push = jest.fn();
  const back = jest.fn();
  jest.spyOn(NextNavigation, "useRouter").mockReturnValue({ replace, push, back } as any);
  return { replace, push, back };
};

export const spyUsePathname = (pathname: string) => {
  jest.spyOn(NextNavigation, "usePathname").mockReturnValue(pathname);
};

export const spyUseSearchParams = (): SpyUseSearchParams => {
  const get = jest.fn();
  jest.spyOn(NextNavigation, "useSearchParams").mockReturnValue({ get } as any);
  return {
    get,
  };
};

export const spyUseServerInsertedHTML = () => {
  return jest.spyOn(NextNavigation, "useServerInsertedHTML");
};
