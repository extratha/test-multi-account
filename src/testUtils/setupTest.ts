import "@testing-library/jest-dom";
import "jest-canvas-mock";

import "@/__mocks__/Markdown";
import "@/__mocks__/MockFirebase";
import "@/__mocks__/NextImage";
import { LOCALE_LANGUAGE } from "@/config";
import { initI18next } from "@/locales/i18n";
import locale from "@/locales/th.json";

initI18next(LOCALE_LANGUAGE.TH, locale);

jest.mock("@/api/api", () => {
  const actualModule = jest.requireActual("@/api/api");
  return { ...actualModule, __esModule: true };
});

jest.mock("@/api/apiUnauthorize", () => {
  const actualModule = jest.requireActual("@/api/apiUnauthorize");
  return { ...actualModule, __esModule: true };
});

jest.mock("next/config", () => () => ({
  setConfig: jest.fn(),
  publicRuntimeConfig: {
    apiToken:
      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE3MDY2NzUzNDQsImlhdCI6MTcwNjU4ODk0NCwicHJvZmlsZUlkIjoiZDU1YWY3NDUtOTYyZS00MTk2LTkyOTEtZDg2YTliYWNiZWY2In0.sNgwYjIGLYEVAptqILGgRVGXwt9FCPKHRubEZyaodhg",
    mockBaseApiUrl: "https://www.mock-playground-dashboard.co.th",
    baseApiUrl: "https://www.mock-playground-dashboard.co.th",
    globalEndpoint: "playground-dashboard",
    publicApiHost: "http://localhost:3000/",
  },
}));

jest.mock("cookies-next", () => ({
  __esModule: true,
  ...jest.requireActual("cookies-next"),
  getCookie: jest.fn().mockReturnValue("MOCK_ACCESS_TOKEN"),
}));
