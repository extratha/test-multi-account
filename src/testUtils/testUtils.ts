import { act, render as testRender } from "@testing-library/react";
import dayjs from "dayjs";
import flushPromises from "flush-promises";
import { ReactElement } from "react";

import { TestProvider } from "./TestProvider";

export const API = {
  LOGIN: "/auth/login",
  EXAMPLE_AI_INTERPRET_URL: "/lab/examples/labId",
  GET_LAB_INTERPRETS_ID_URL: "/lab/interprets/labId",
  SUBMIT_LAB_INTERPRETS_URL: "/lab/interprets",
  CONSENT: "/consents",
  TERMS_AND_CONDITIONS: "/consents/terms-conditions",
  TERMS_AND_CONDITIONS_LATEST: "/consents/terms-conditions/latest",
  PRIVACY_POLICY: "/consents/privacy-policies",
  PRIVACY_POLICY_LATEST: "/consents/privacy-policies/latest",
  PUBLIC_TERMS_CONDITIONS: "/consents/terms-conditions/public",
  PUBLIC_PRIVACY_POLICIES: "/consents/privacy-policies/public",
};

export const render = (ui: ReactElement) => {
  return testRender(ui, { wrapper: TestProvider });
};

export const flushPromise = async () => {
  await act(async () => {
    await flushPromises();
  });
};

export const advanceTimersByTime = async (time: number) => {
  await act(async () => {
    jest.advanceTimersByTime(time);
  });
};

export const setMockDate = (date: string) => {
  jest.setSystemTime(dayjs(date).valueOf());
};
