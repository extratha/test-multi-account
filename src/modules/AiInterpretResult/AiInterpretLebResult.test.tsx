import { render, userEvent, screen } from "@/__tests__/testUtils";
import AiInterpretLabResult, { AiInterpretLabResultProps } from "./AiInterpretLebResult";

describe("AiInterpretLabResult", () => {
  let props: AiInterpretLabResultProps;

  beforeEach(() => {
    props = {
      name: "lab",
      group: {
        groupName: "lipidProfile",
        data: [
          {
            key: "cholesterol_value",
            value: "208",
            unit: "mg/dL",
            range: [
              {
                value: "-",
                description: "abnormal",
              },
            ],
          },
        ],
      },
    };
  });

  const renderLabUnit = () => {
    return render(<AiInterpretLabResult {...props} />);
  };

  it("should render correctly", () => {
    const { asFragment } = renderLabUnit();
    expect(asFragment()).toMatchSnapshot();
  });

  it("should show result lab when click button collapse", async () => {
    renderLabUnit();

    expect(screen.getByTestId("lab-lipidProfile-cholesterol_value-label")).not.toBeVisible();

    await userEvent.click(screen.getByTestId("lab-lipidProfile-button-collapse"));

    expect(screen.getByTestId("lab-lipidProfile-cholesterol_value-label")).toBeVisible();
  });
});
