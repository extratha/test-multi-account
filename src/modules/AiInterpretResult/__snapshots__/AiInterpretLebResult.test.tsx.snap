// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AiInterpretLabResult should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #eceff1;
  border-radius: 8px;
  overflow: hidden;
}

.emotion-0 .MuiButton-root {
  font-size: 12px;
  font-weight: 600;
  color: #FFFFFF;
  background-color: #212121;
}

.emotion-0 .MuiButton-root:hover {
  background-color: #212121;
}

.emotion-0 .MuiButton-endIcon {
  padding-left: 8px;
}

.emotion-1 {
  position: absolute;
}

.emotion-2 {
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 24px 12px 12px 40px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  background-color: #FAFAFA;
}

.emotion-3 {
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-4 {
  margin: 0;
  font-size: 18px;
  font-weight: 700;
  line-height: 24px;
}

.emotion-5 {
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 8px;
  padding: 8px 14px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #2196F3;
  font-family: fontFamily;
  font-weight: 700;
  font-size: 12px;
  line-height: 1.5;
  text-transform: uppercase;
  min-width: 64px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  box-shadow: 0px 4px 6px -1px #21212133,0px 2px 4px -2px #21212133;
}

.emotion-5::-moz-focus-inner {
  border-style: none;
}

.emotion-5.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-5 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-5:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(33, 150, 243, 0.04);
}

@media (hover: none) {
  .emotion-5:hover {
    background-color: transparent;
  }
}

.emotion-5.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-5 svg {
  width: 16px;
  height: 16px;
}

.emotion-6 {
  display: inherit;
  margin-right: -4px;
  margin-left: 8px;
}

.emotion-6>*:nth-of-type(1) {
  font-size: 20px;
}

.emotion-7 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-8 {
  height: 0;
  overflow: hidden;
  -webkit-transition: height 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: height 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  visibility: hidden;
}

.emotion-9 {
  display: flex;
  width: 100%;
}

.emotion-10 {
  width: 100%;
}

.emotion-11 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 12px 24px;
}

.emotion-11>:not(style):not(style) {
  margin: 0;
}

.emotion-11>:not(style)~:not(style) {
  margin-top: 16px;
}

.emotion-12 {
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-13 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.emotion-13>:not(style):not(style) {
  margin: 0;
}

.emotion-13>:not(style)~:not(style) {
  margin-top: 8px;
}

.emotion-14 {
  margin: 0;
  font-size: 16px;
  font-weight: 700;
  line-height: 1.5;
}

.emotion-15 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 25%;
}

.emotion-16 {
  margin: 0;
  font-size: 16px;
  font-weight: 700;
  line-height: 1.5;
  text-align: end;
}

.emotion-17 {
  margin: 0;
  font-size: 14px;
  line-height: 1.5;
  text-align: end;
}

<div
    class="MuiStack-root emotion-0"
  >
    <svg
      class="emotion-1"
    />
    <div
      class="MuiStack-root emotion-2"
    >
      <div
        class="MuiStack-root emotion-3"
      >
        <p
          class="MuiTypography-root MuiTypography-titleBold emotion-4"
          data-testid="lab-title-lipidProfile"
        >
          ระดับไขมันในเลือด
        </p>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary emotion-5"
        data-testid="lab-lipidProfile-button-collapse"
        tabindex="0"
        type="button"
      >
        ดูผล
        <span
          class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium emotion-6"
        >
          <svg />
        </span>
        <span
          class="MuiTouchRipple-root emotion-7"
        />
      </button>
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-8"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical emotion-9"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-10"
        >
          <div
            class="MuiStack-root emotion-11"
          >
            <div
              class="MuiStack-root emotion-12"
            >
              <div
                class="MuiStack-root emotion-13"
              >
                <p
                  class="MuiTypography-root MuiTypography-bodyBold emotion-14"
                  data-testid="lab-lipidProfile-cholesterol_value-label"
                >
                  Cholesterol
                </p>
              </div>
              <div
                class="MuiStack-root emotion-15"
              >
                <p
                  class="MuiTypography-root MuiTypography-bodyBold emotion-16"
                  data-testid="lab-lipidProfile-cholesterol_value-value"
                >
                  208 (mg/dL)
                </p>
                <p
                  class="MuiTypography-root MuiTypography-bodySmall emotion-17"
                  data-testid="lab-lipidProfile-cholesterol_value-unit"
                >
                  (- mg/dL)
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
